docker-compose.yml

```diff
@@ -9,39 +9,66 @@ services:
       - db-data:/var/lib/postgresql/data
     environment:
       POSTGRES_PASSWORD: postgres
+    healthcheck:
+      test: ['CMD-SHELL', 'pg_isready -U postgres']
+      interval: 5s
+      timeout: 5s
+      retries: 5
 
   subquery-node:
     build: ./
     extra_hosts:
       - 'host.docker.internal:host-gateway'
     depends_on:
-      - 'postgres'
+      'postgres':
+        condition: service_healthy
     restart: always
     environment:
       DB_USER: postgres
       DB_PASS: postgres
       DB_DATABASE: postgres
-      NODE_ENV: local
       DB_HOST: postgres
       DB_PORT: 5432
-      # START_BLOCK: 30596
-      # NETWORK_ENDPOINT: wss://testnet-rpc.polymesh.live
-      # NO_NATIVE_GRAPHQL_DATA: 'TRUE'
-      CALL_HANDLER: 'handleCall' # needed to generate project.yaml
-      EVENT_HANDLER: 'handleEvent' # needed to generate project.yaml
+      NODE_ENV: local
+      START_BLOCK: 1
+      NO_NATIVE_GRAPHQL_DATA: 'TRUE' # Setting this value only processes tooling specific handlers. Native subquery entities are ignored.
+      # Polymesh-local
       NETWORK_ENDPOINT: ws://host.docker.internal:9944
+      NETWORK_CHAIN_ID: '0xda7f2072787bfd0b09f7e12fca619afb6041b3d620f39f3a508814869100bf01'
+      # Staging
+      # NETWORK_ENDPOINT: wss://staging-rpc.polymesh.live
+      # NETWORK_CHAIN_ID: '0x3c3183f6d701500766ff7d147b79c4f10014a095eaaa98e960dcef6b3ead50ee'
+      # Testnet
+      # NETWORK_ENDPOINT: wss://testnet-rpc.polymesh.live
+      # NETWORK_CHAIN_ID: '0x2ace05e703aa50b48c0ccccfc8b424f7aab9a1e2c424ed12e45d20b1e8ffd0d6'
+      # Mainnet
+      # NETWORK_ENDPOINT: wss://mainnet-rpc.polymesh.network
+      # NETWORK_CHAIN_ID: '0x6fbd74e5e1d0a61d52ccfe9d4adaed16dd3a7caa37c6bc4d0c2fa12e8b2f4063'
+    volumes:
+      - ./:/app
     command:
-      - --batch-size=500
-      - -f=/app
-      - --local
+      - -f=/app # Path of the project
+      - --batch-size=100 # Batch size of blocks to fetch in one round
+      # - --scale-batch-size # scale batch size based on memory usage. To know more about the process - https://github.com/subquery/subql/blob/c9bc9733deef726f78fa48387e74e52f8d6ca8d2/packages/node/src/indexer/fetch.service.ts#L88
+      # - --timeout=3600 # Timeout for indexer sandbox to execute the mapping functions (in seconds)
```

Comment:
OK to leave, but an hour seems like quite a bit to me.

My preference is to have realistic values as a bit of documentation and suggestion for what config values should be. Not sure what it needs to be to deal with a large block though.

Reply:
I wrote that values thinking that SRE team won't have think too much if they are asked to run with any params.

As for the large block, reducing batch-size to 1 and setting timeout to 3600 worked for me. When Robert does those batch transactions we could recheck on what those block need.

Reply:
My concern is along the lines the connection dies (or something similar) and SQ falls behind a whole hour before reconnecting.

I think we are OK reducing the batch size down to something like 100 without impacting the sync time to much, since we need to wait for a SQL commit on every block. Some SubQuery projects can skip block entirely so probably benefit more from bigger batch sizes. I think 1 is too small and will slow down sync though.

Might be worth experimenting.